goagen failed with required UUID header

I found bootstrap failed with UUID type and Required.

Example design is here.

```
package design

import (
	. "github.com/goadesign/goa/design"
	. "github.com/goadesign/goa/design/apidsl"
)

var _ = API("sample", func() {
	Title("sample")
	Description("sample")
	Scheme("http")
	Host("localhost:8080")
})

var _ = Resource("tests", func() {
	BasePath("/tests")
	DefaultMedia(TestMedia)

	Action("list", func() {
		Description("")
		Routing(
			GET("/"),
		)
		Headers(func() {
			Header("X-UUID", UUID, "UUID")
			Required(
				"X-UUID",
			)
		})
		Response(OK)
		Response(NotFound)
	})
})

var TestMedia = MediaType("application/vnd.sample.v1+json", func() {
	Description("video")

	Attributes(func() {
		Attribute("id")
	})

	View("default", func() {
		Attribute("id")
	})
})
```

and displays erros is here.
```
goagen bootstrap -d <GOPATH>/goatest/design
exit status 1
/***/goatest/client/tests.go:46:29: expected ';', found 'return'

========
Content:
package client

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"io/ioutil"
	"net/http"
	"net/url"
	"os"
	"path"
	"strconv"
	"strings"
	"time"
	"golang.org/x/net/context"
	"golang.org/x/net/websocket"
	uuid "github.com/goadesign/goa/uuid"
)

// ListTestsPath computes a request path to the list action of tests.
func ListTestsPath() string {
	return fmt.Sprintf("/tests", )
}
// ListTests makes a request to the list action endpoint of the tests resource
func (c *Client) ListTests(ctx context.Context, path string, xUUID uuid.UUID) (*http.Response, error) {
	req, err := c.NewListTestsRequest(ctx, path, xUUID)
	if err != nil {
		return nil, err
	}
	return c.Client.Do(ctx, req)
}
// NewListTestsRequest create the request corresponding to the list action endpoint of the tests resource.
func (c *Client) NewListTestsRequest(ctx context.Context, path string, xUUID uuid.UUID) (*http.Request, error) {
	scheme := c.Scheme
	if scheme == "" {
		scheme = "http"
	}
	u := url.URL{Host: c.Host, Scheme: scheme, Path: path}
	req, err := http.NewRequest("GET", u.String(), nil)
	if err != nil {
		return nil, err
	}
	header := req.Header
	tmp3 := xUUID.String()
	header.Set("X-UUID", tmp3)	return req, nil
}
```

Thanks.

--- a/goagen/gen_client/generator.go
+++ b/goagen/gen_client/generator.go
@@ -1005,8 +1005,8 @@ func (c *Client) {{ $funcName }}(ctx context.Context, path string{{ if .Params }
 {{ end }}{{ if .MustToString }}{{ $tmp := tempvar }}	{{ toString .ValueName $tmp .Attribute }}
 	header.Set("{{ .Name }}", {{ $tmp }}){{ else }}
 	header.Set("{{ .Name }}", {{ .ValueName }})
-{{ end }}{{ if .CheckNil }}	}
-{{ end }}{{ end }}{{ end }}{{ if .Signer }}	if c.{{ .Signer }}Signer != nil {
+{{ end }}{{ if .CheckNil }}	}{{ end }}
+{{ end }}{{ end }}{{ if .Signer }}	if c.{{ .Signer }}Signer != nil {
 		c.{{ .Signer }}Signer.Sign(req)
 	}
 {{ end }}	return req, nil

--- a/goagen/gen_client/generator_test.go
+++ b/goagen/gen_client/generator_test.go
@@ -7,6 +7,7 @@ import (
 	"strings"
 
 	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/dslengine"
 	"github.com/goadesign/goa/goagen/codegen"
 	"github.com/goadesign/goa/goagen/gen_client"
 	"github.com/goadesign/goa/version"
@@ -36,6 +37,53 @@ var _ = Describe("Generate", func() {
 	AfterEach(func() {
 		os.RemoveAll(outDir)
 		delete(codegen.Reserved, "client")
+	})
+
+	Context("with a required UUID header", func() {
+		BeforeEach(func() {
+			codegen.TempCount = 0
+			o := design.Object{
+				"header_name": &design.AttributeDefinition{Type: design.UUID},
+			}
+			design.Design = &design.APIDefinition{
+				Name: "testapi",
+				Resources: map[string]*design.ResourceDefinition{
+					"foo": {
+						Name: "foo",
+						Actions: map[string]*design.ActionDefinition{
+							"show": {
+								Name: "show",
+								Routes: []*design.RouteDefinition{
+									{
+										Verb: "GET",
+										Path: "",
+									},
+								},
+								Headers: &design.AttributeDefinition{
+									Type: o,
+									Validation: &dslengine.ValidationDefinition{
+										Required: []string{"header_name"},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			fooRes := design.Design.Resources["foo"]
+			showAct := fooRes.Actions["show"]
+			showAct.Parent = fooRes
+			showAct.Routes[0].Parent = showAct
+		})
+
+		It("generates header initialization code that compiles", func() {
+			Ω(genErr).Should(BeNil())
+			Ω(files).Should(HaveLen(9))
+			c, err := ioutil.ReadFile(filepath.Join(outDir, "client", "foo.go"))
+			Ω(err).ShouldNot(HaveOccurred())
+			content := string(c)
+			Ω(content).Should(ContainSubstring("header.Set(\"header_name\", tmp3)\n"))
+		})
 	})
 
 	Context("with jsonapi like querystring params", func() {

